ci/jenkins/test.sh

```diff
@@ -984,6 +994,7 @@ if [[ $TESTCASE =~ "multicast" ]]; then
 fi
 
 source $WORKSPACE/ci/jenkins/utils.sh
+docker_login "${dockerUser}" "${dockerPassword}"
```

Comment:
Not sure if there is such convention: should it disable xtrace first before using the secret, otherwise it would be printed to terminal by enabling xtrace in other place in this script and it won't be easy to notice.
The following code should ensure it won't be printed:
```bash
(
  set +x 
  docker_login "${dockerUser}" "${dockerPassword}"
)
```
@antoninbas do you know how is this usually handled?

Reply:
The secret is stored in Jenkins credentials and will not be exposed in the results. However, we should indeed avoid printing too much information about the login step, will update the code.

Reply:
Personally I think like this should not even be in the script if possible.
Isn't there a way to configure Jenkins so that we login to Docker before even calling this script?

Reply:
@antoninbas Do you mean to put this login script into Jenkins macros to avoid modifying the test.sh related scripts? Using this method, we would need to modify the macros definition for all related jobs. IMO this might actually be more complex than directly inserting the script into test.sh, because we would need to modify all the relevant macros. Also, in the Nephe CI pipeline, we have verified a similar method to log into Docker as we did in this PR.  https://github.com/antrea-io/nephe/blob/9faeba2c189821045e4ea6d37a1c97dbdecf67bd/ci/jenkins/nephe-ci.sh#L112

Reply:
> Personally I think like this should not even be in the script if possible. Isn't there a way to configure Jenkins so that we login to Docker before even calling this script?

Yes, Handling Docker login credentials within a script can be a security risk. It's generally better to manage authentication outside of the script . isn't the docker credentials be stored in jenkins credentials and used to access it using credentials in job itself, In that case it has be configured in each and every job then !

Reply:
> > Personally I think like this should not even be in the script if possible. Isn't there a way to configure Jenkins so that we login to Docker before even calling this script?
> 
> Yes, Handling Docker login credentials within a script can be a security risk. It's generally better to manage authentication outside of the script . isn't the docker credentials be stored in jenkins credentials and used to access it using credentials in job itself, In that case it has be configured in each and every job then !

When using the Jenkins Credential secrets, you can securely inject sensitive credentials into the build process as environment variables. These variables are visible during use, but Jenkins automatically masks their values to prevent them from being displayed in the build logs.

After consideration, I placed the login script in the macro YAML, which ensures that the login step cannot be changed by any PR.